### PR TITLE
Remove Qthreads support

### PR DESCRIPTION
**Pull request by masterleinad**

### CODE REVIEW

**Comment by masterleinad** [comment]
Retest this please.

File: core/src/Kokkos_Core_fwd.hpp [hunk 1]
```diff
@@ -171,38 +167,34 @@ typedef Experimental::ROCm DefaultExecutionSpace;
...
-//#elif defined( KOKKOS_ENABLE_DEFAULT_DEVICE_TYPE_QTHREADS )
-//  typedef Qthreads DefaultExecutionSpace;
 #elif defined(KOKKOS_ENABLE_DEFAULT_DEVICE_TYPE_HPX)
 typedef Kokkos::Experimental::HPX DefaultExecutionSpace;
 #elif defined(KOKKOS_ENABLE_DEFAULT_DEVICE_TYPE_SERIAL)
 typedef Serial DefaultExecutionSpace;
 #else
 #error \
-    "At least one of the following execution spaces must be defined in order to use Kokkos: Kokkos::Cuda, Kokkos::Experimental::OpenMPTarget, Kokkos::OpenMP, Kokkos::Threads, Kokkos::Qthreads, or Kokkos::Serial."
+    "At least one of the following execution spaces must be defined in order to use Kokkos: Kokkos::Cuda, Kokkos::Experimental::OpenMPTarget, Kokkos::Experimental::HIP, Kokkos::OpenMP, Kokkos::Threads, Kokkos::Experimental::HPX, or Kokkos::Serial."
```
Comment (dalg24) [inline core/src/Kokkos_Core_fwd.hpp]:
```suggestion
    "At least one of the following execution spaces must be defined in order to use Kokkos: Kokkos::Cuda, Kokkos::Experimental::HIP, Kokkos::Experimental::OpenMPTarget, Kokkos::OpenMP, Kokkos::Threads, Kokkos::Experimental::HPX, or Kokkos::Serial."
```

**Review by dalg24 — APPROVED** [review]

File: core/src/Kokkos_Core_fwd.hpp (hunk 1 above)
Comment (dalg24) [reply core/src/Kokkos_Core_fwd.hpp]: (I was just kidding but yes we should prioritize Hip over OpenMP Target)

Comment (masterleinad) [reply core/src/Kokkos_Core_fwd.hpp]: I think that was a sensible suggestion.

There are still a number of places where we don't list the execution spaces (in particular the new ones) consistently. We should look into this later.

**Pull request merged**